api/v1alpha1/externaldns_webhook.go

```diff
@@ -17,60 +17,26 @@ limitations under the License.
 package v1alpha1
 
 import (
-	"context"
 	"errors"
 	"fmt"
-
-	configv1 "github.com/openshift/api/config/v1"
-
-	"k8s.io/apimachinery/pkg/types"
-
 	"regexp"
 
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/client-go/discovery"
-	"k8s.io/client-go/kubernetes"
 
 	utilErrors "k8s.io/apimachinery/pkg/util/errors"
 	ctrl "sigs.k8s.io/controller-runtime"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
 )
 
-const (
-	kind    = "OpenShiftAPIServer"
-	group   = "operator.openshift.io"
-	version = "v1"
-)
-
 // webhookLog is for logging in this package.
 var webhookLog = logf.Log.WithName("validating-webhook")
 
-var IsOpenShift bool
+var isOpenShift bool
 
-var PlatformStatus *configv1.PlatformStatus
-
-func (r *ExternalDNS) SetupWebhookWithManager(mgr ctrl.Manager) error {
-	kubeClient, err := kubernetes.NewForConfig(mgr.GetConfig())
-	if err != nil {
-		return err
-	}
-	IsOpenShift = IsOCP(kubeClient)
-
-	if IsOpenShift {
-
-		var err error
-		infraConfig := &configv1.Infrastructure{}
-		if err = mgr.GetClient().Get(context.TODO(), types.NamespacedName{Name: "cluster"}, infraConfig); err != nil {
-			return fmt.Errorf("failed to get infrastructure 'config': %v", err)
-		}
-
-		PlatformStatus = infraConfig.Status.PlatformStatus
-	}
-
-	return ctrl.NewWebhookManagedBy(mgr).
-		For(r).
-		Complete()
+func (r *ExternalDNS) SetupWebhookWithManager(mgr ctrl.Manager, openshift bool) error {
```

Comment:
It may be time to end this hacky way in which kubebuilder insists on registering the webhook. In a PR once this is merged I think I would like to implement a struct which embeds `ExternalDNS` and has the `isOpenshift` field. This way we can avoid the package level variable here as well.